config/src/config/mod.rs

```diff
@@ -110,6 +110,25 @@ impl BaseConfig {
             file_path.clone()
         }
     }
+
+    /// Returns true if the default is set, the current file path is empty, and the default file
+    /// path points to an existing file.
+    pub fn test_and_set_full_path<T>(
+        &self,
+        config_value: &T,
+        default_value: &T,
+        config_path: &mut PathBuf,
+        default_path: &str,
```

Comment:
why do we test both config_path and config_value? Can't we just check that config_path is not set?